### PR TITLE
expose techInsightsApiRef

### PR DESCRIPTION
**Pull request by goenning**
Signed-off-by: goenning <me@goenning.net>

## Hey, I just made a Pull Request!

I want to create a custom page which uses tech-insights API and I want to do something like the following, but the apiRef was not publicly exposed. I had a look on other packages and they usually expose it, so hence this PR!

```ts
  const api = useApi(techInsightsApiRef);
  const { namespace, kind, name } = useParams();

  const { value, loading, error } = useAsync(
    async () => await api.runChecks({ namespace, kind, name }),
  );
```

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/tech-insights/api-report.md [hunk 1]
```diff
@@ -5,12 +5,22 @@
...
+import { CheckResult } from '@backstage/plugin-tech-insights-common';
+import { EntityName } from '@backstage/catalog-model';
+import { default as React_2 } from 'react';
 import { RouteRef } from '@backstage/core-plugin-api';
 
 // @public (undocumented)
 export const EntityTechInsightsScorecardContent: () => JSX.Element;
 
+// Warning: (ae-forgotten-export) The symbol "TechInsightsApi" needs to be exported by the entry point index.d.ts
+// Warning: (ae-missing-release-tag) "techInsightsApiRef" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
```
Comment (Rugvip) [inline plugins/tech-insights/api-report.md:19]: Let's make sure we're not adding new warnings to the API report, which would involve TsDoc comments with a `@public` tag where needed, and making sure types are exported as needed.

File: .changeset/slimy-socks-pump.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-tech-insights': patch
+---
+
+expose apiRef
```
Comment (Rugvip) [inline .changeset/slimy-socks-pump.md:5]:
```suggestion
Export `techInsightsApiRef` and associated types.
```

**Review by Rugvip — COMMENTED** [review]
Thanks! 👍 

Just some tidying up :p

**Comment by Rugvip** [comment]
Btw, I'm also assuming this is something we want to do atm. Was the export avoided for a particular reason @Irma12 or is this fine?

**Review by Irma12 — APPROVED** [review]

**Comment by Irma12** [comment]
> Btw, I'm also assuming this is something we want to do atm. Was the export avoided for a particular reason @Irma12 or is this fine?

No particular reason to be honest, so I think it is fine to export it @Rugvip

**Review by Irma12 — APPROVED** [review]

**Comment by goenning** [comment]
@Rugvip @Irma12 please have another look. Thanks

side question: is it possible to run api-reports on a single plugin? running it on root folder is slow.

**Comment by Rugvip** [comment]
@goenning nop, the main chunk of it is `tsc`, and it has to be run with `--incremental false` and `--skipLibCheck false` or you end up with variations between runs :<

**Review by Rugvip — APPROVED** [review]
Nice, thanks! 👌

**Pull request merged**